lib/DefaultResolver.php

```diff
@@ -214,6 +218,8 @@ private function doResolve($name, array $types, $options) {
             $this->config = (yield \Amp\resolve($this->loadResolvConf()));
         }
 
+        $config = $this->loadEnvConfig();
```

Comment:
Yes, it might change. And if it changes, previous settings from `/etc/resolv.conf` + `RES_OPTIONS` should apply instead of `/etc/resolv.conf` + previous `RES_OPTIONS` + `RES_OPTIONS`.